File: solutions/problem_1466_reorder_routes_to_make_all_paths_lead_to_the_city_zero/solution.py
"""
Problem 1466: Reorder Routes to Make All Paths Lead to the City Zero
==================================================================
Difficulty: Medium
Tags: Depth-First Search, Breadth-First Search, Graph Theory
Pattern: DFS Graph Traversal

Time Complexity:  O(V + E)
Space Complexity: O(V)
"""

class Solution:
    def minReorder(self, n: int, connections: List[List[int]]) -> int:
        # DFS on graph - O(V+E) time
        visited = set()
        result = []
        
        def dfs(node):
            if node in visited:
                return
            visited.add(node)
            result.append(node)
            # Traverse neighbors (adjust based on adjacency representation)
        
        dfs(0)
        return result if isinstance(0, list) else len(result)

File: solutions/problem_1466_reorder_routes_to_make_all_paths_lead_to_the_city_zero/solution.cpp
/*
 * Problem 1466: Reorder Routes to Make All Paths Lead to the City Zero
 * ==================================================================
 * Difficulty: Medium
 * Tags: Depth-First Search, Breadth-First Search, Graph Theory
 * Pattern: DFS Graph Traversal
 *
 * Time Complexity:  O(V + E)
 * Space Complexity: O(V)
 */

#include <functional>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int minReorder(int n, vector<vector<int>>& connections) {
        // DFS on graph - O(V+E) time
        vector<bool> visited(n.size(), false);
        vector<int> result;
        function<void(int)> dfs = [&](int node) {
            if (visited[node]) return;
            visited[node] = true;
            result.push_back(node);
            // Traverse neighbors
        };
        dfs(0);
        return result;
    }
};
